testonly/hammer/hammer.go

```diff
@@ -181,15 +185,58 @@ func HitMap(cfg MapConfig) error {
 		}
 	}(ticker.C)
 
-	for count := uint64(1); count < cfg.Operations; count++ {
-		if err := s.retryOneOp(ctx); err != nil {
-			return err
+	var wg sync.WaitGroup
+	errs := make(chan error, cfg.NumCheckers)
+
+	cctx, cancel := context.WithCancel(ctx)
```

Comment:
you only really use the channel from `cctx`, could you move `done` up to here from where it's currently created below, pass it into `readChecker`, remove the `done <- true` and add a `defer close(done)` to the top of that goroutine?
Then you can remove `cancel()` and `close(done)` from L235 and L238.

Reply:
Done.